### DISCUSSION TITLE
Avalonia unable to get members of Vector2?

### DISCUSSION THREAD

**Discussion by Mrcubix** [AvaloniaUI/Avalonia#13234] [Q&A]
Currently stuck with following exception when trying to provide the size of the shape in a canvas.

```log
Avalonia error AVLN:0004: Unable to resolve property or method of name 'X' on type 'System.Numerics.Vector2'
 ```

The shape control in question is used as follow:

```axaml
<gn:RectangleNode Width="{Binding Size.X}" Height="{Binding Size.Y}" 
                  Canvas.Left="{Binding Position.X}" Canvas.Top="{Binding Position.Y}" />
```

Compiled Bindings are enabled.

The reason I'm using Vector2 for consistency since the app I'm connecting to using pipes use Vector2.

**Comment by timunie** [comment]
I wonder if these are actually properties or Fields only. Fields are not accessible from XAML.

**Reply by Mrcubix** [reply]
Seems like it's a public field, guess I'll have to use yet another workaround just to make it work with Avalonia

**Reply by Mrcubix** [reply]
Nvm none of the types i have checked in Avalonia use properties for the X, Y, Width or Height values, anything that can be done without having to remake 6 different types from scratch?

(No, `Point` & `Size` does not seem to use property as the same exception occur)

**Reply by maxkatz6** [reply]
Since it's not a mutable struct, you can use converters.

**Reply by maxkatz6** [reply]
Avalonia structs usually have properties instead of public fields, and it work with bindings. Typical binding use case - control.Bounds.Width/Bounds.Height
